### PR TITLE
test: add Qwen-Image-Edit-2511 accuracy e2e

### PR DESCRIPTION
**Pull request by david6666666**
## Purpose

Add a dedicated Qwen-Image-Edit-2511 accuracy test suite under `tests/e2e/accuracy/qwen_image_edit_2511`.

This PR adds:
- a diffusers baseline runner for `Qwen/Qwen-Image-Edit-2511`
- an online-serving accuracy test that exercises `/v1/images/edits` with `omni_server`
- a pixel-level comparison test that generates artifacts and summary JSON for offline vs online outputs
- Buildkite nightly coverage in `.buildkite/test-nightly-diffusion.yml`
- image source adaptation for remote URLs, local paths, and data URLs

Note: the provided baseline snippet referenced `image2` without defining a second source, so this change defaults to one source but supports repeated `--qwen-image-edit-2511-image-source` inputs for multi-image cases.

## Test Plan

Local static validation:
- `python -m py_compile tests/e2e/accuracy/conftest.py tests/e2e/accuracy/qwen_image_edit_2511/__init__.py tests/e2e/accuracy/qwen_image_edit_2511/run_qwen_image_edit_2511_diffusers_baseline.py tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_online_serving.py tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_pixel_similarity.py`
- `git diff --check`

Remote/nightly validation target:
- `pytest -s -v tests/e2e/accuracy/qwen_image_edit_2511 -m "advanced_model and benchmark and diffusion and H100" --run-level advanced_model`

## Test Result

Local results:
- `python -m py_compile ...`: PASS
- `git diff --check`: PASS
- `pytest --collect-only tests/e2e/accuracy/qwen_image_edit_2511 -q`: blocked locally because repo-level `tests/conftest.py` imports `cv2`, and this environment does not have `cv2` installed

```
=========================================================================== warnings summary ===========================================================================
../.venv/lib/python3.12/site-packages/transformers/utils/hub.py:110
  /mnt/data4/cwq/.venv/lib/python3.12/site-packages/transformers/utils/hub.py:110: FutureWarning: Using `TRANSFORMERS_CACHE` is deprecated and will be removed in v5 of Transformers. Use `HF_HOME` instead.
    warnings.warn(

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

../.venv/lib/python3.12/site-packages/torch/jit/_script.py:362: 14 warnings
  /mnt/data4/cwq/.venv/lib/python3.12/site-packages/torch/jit/_script.py:362: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

../.venv/lib/python3.12/site-packages/_pytest/config/__init__.py:1434
  /mnt/data4/cwq/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: asyncio_mode
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================== 1 passed, 18 warnings in 250.84s (0:04:10) ==============================================================
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
```
Remote H100 execution was not run locally in this environment.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [x] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>**

### CODE REVIEW

**Comment by fhfuih** [comment]
Ooops I also had one WIP branch at https://github.com/fhfuih/vllm-omni/tree/qwen-image-edit-acc before the holiday.

Mine also include Qwen Image Edit (not 2511)

At a quick glance, I think yours has a more complete metric set. But some utility functions may be put in a shared file because they are quite likely to be reused by other tests.

File: .buildkite/test-nightly-diffusion.yml [hunk 1]
```diff
@@ -319,6 +319,49 @@ steps:
...
+        agents:
+          queue: "mithril-h100-pool"
+        plugins:
+          - kubernetes:
+              podSpec:
+                containers:
+                  - image: 936637512419.dkr.ecr.us-west-2.amazonaws.com/vllm-ci-pull-through-cache/q9t5s3a7/vllm-ci-test-repo:$BUILDKITE_COMMIT
+                    resources:
+                      limits:
+                        nvidia.com/gpu: 2
```
Comment (fhfuih) [inline .buildkite/test-nightly-diffusion.yml:338]: Without any parallelism, we can request only one GPU

File: tests/e2e/accuracy/qwen_image_edit_2511/__init__.py
Comment (fhfuih) [inline tests/e2e/accuracy/qwen_image_edit_2511/__init__.py:1]: I think a lot of functions in this file can be put in a separate `tests/e2e/accuracy/utils.py` or `tests/e2e/accuracy/conftest.py` (if it can be written as a fixture).

This way, actual test files won't do `from tests.e2e.accuracy.qwen_image_edit_2511 import ...` from *within* the `tests/e2e/accuracy/qwen_image_edit_2511` folder, which seems quite weird 😂

File: tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_online_serving.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    )
+    return paths["online"]
+
+
+@pytest.mark.advanced_model
+@pytest.mark.benchmark
+@pytest.mark.diffusion
+@hardware_test(res={"cuda": "H100"}, num_cards=2)
+@pytest.mark.parametrize("omni_server", SERVER_CASES, indirect=True)
+def test_qwen_image_edit_2511_online_serving_generates_image(
```
Comment (fhfuih) [inline tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_online_serving.py:115]: I don't think we need a basic test of output dimensions here. They should have been covered already by any other L4 functionality test or L2~L3 e2e test that uses `omni_server` and `openai_client.send_diffusion_request`. It does the same basic check internally

File: tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_pixel_similarity.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        accuracy_artifact_root=accuracy_artifact_root,
+    )
+
+    online_image = Image.open(online_path)
+    online_image.load()
+    offline_image = Image.open(offline_path)
+    offline_image.load()
+
+    assert online_image.size == (WIDTH, HEIGHT)
+    assert offline_image.size == (WIDTH, HEIGHT)
```
Comment (fhfuih) [inline tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_pixel_similarity.py]: In [my other WIP branch](https://github.com/fhfuih/vllm-omni/commit/a5b663d3e591111e88e21e08d3db29f89ef1f130#diff-8bf35941c7a69840a9f08b7dbe8f3ad5f6d8c5fd736bb58dd034761bc86af1a5), my logic is that: if diffusers output cannot even pass basic size check, then `pytest.skip` this test case. We consider the baseline as broken and hence the comparison not even needed.

See if you think this is reasonable and want to add it here

File: tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_pixel_similarity.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    ]
+    existing_pythonpath = env.get("PYTHONPATH")
+    if existing_pythonpath:
+        pythonpath_parts.append(existing_pythonpath)
+    env["PYTHONPATH"] = os.pathsep.join(pythonpath_parts)
+    env.setdefault("VLLM_WORKER_MULTIPROC_METHOD", "spawn")
+    return env
+
+
+def _build_diffusers_command(
```
Comment (fhfuih) [inline tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_pixel_similarity.py:42]: I think these diffusers command can be also moved to a shared file

File: tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_pixel_similarity.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+@pytest.mark.diffusion
+@hardware_test(res={"cuda": "H100"}, num_cards=2)
+@pytest.mark.parametrize("omni_server", SERVER_CASES, indirect=True)
+def test_qwen_image_edit_2511_serving_matches_diffusers_pixel_similarity(
+    omni_server: OmniServer,
+    accuracy_artifact_root: Path,
+    qwen_image_edit_2511_image_sources: list[str] | None,
+) -> None:
+    if not torch.cuda.is_available() or torch.cuda.device_count() < 2:
+        pytest.skip("Qwen-Image-Edit-2511 pixel similarity test requires >= 2 CUDA GPUs.")
```
Comment (fhfuih) [inline tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_pixel_similarity.py]: Why? Unless we run diffusers and vllm-omni at the same time, one in cuda:0 and one in cuda:1. But I see that the current test run two libraries sequentially.

File: tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_pixel_similarity.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        command,
+        cwd=REPO_ROOT,
+        env=_runner_env(),
+        check=True,
+        timeout=IMAGE_TIMEOUT_SECONDS,
+    )
+    return paths["offline"]
+
+
+def _diff_metrics(a: Image.Image, b: Image.Image) -> dict[str, float]:
```
Comment (fhfuih) [inline tests/e2e/accuracy/qwen_image_edit_2511/test_qwen_image_edit_2511_pixel_similarity.py]: And these diff metric computation can be moved to a shared file. It should greatly standardize the metric computation for other model tests

**Review by fhfuih — COMMENTED** [review]
Left some comments for your consideration~

**Comment by hsliuustc0106** [comment]
🔴 Pre-commit gate failing. Fix before requesting review.

Test-only PR: Qwen-Image-Edit-2511 accuracy e2e with diffusers baseline. Test plan includes local static validation and remote H100 target. Pre-commit fails. Build passes.

Once gates pass, consider adding actual e2e validation results (Test Result blocked by missing cv2 in local env).

**Comment by Gaohan123** [comment]
Why the PR is closed?

**Comment by fhfuih** [comment]
> Why the PR is closed?

My other PR #2761 does the same thing. After our discussion, we decide to follow my PR because it is cleaner. @david6666666 later also rewrote #2772 for other Qwen Image-series models in the same pattern
